aiida/orm/querybuilder.py

```diff
@@ -491,25 +506,82 @@ def __init__(
         if order_by:
             self.order_by(order_by)
 
-    def __str__(self) -> str:
-        """
-        When somebody hits: print(QueryBuilder) or print(str(QueryBuilder))
-        I want to print the SQL-query. Because it looks cool...
+    def as_dict(self) -> QueryDict:
+        """Convert to a JSON serialisable dictionary representation of the query."""
+        return copy.deepcopy({  # type: ignore[return-value]
+            'path': self._path,
+            'filters': self._filters,
+            'project': self._projections,
+            'order_by': self._order_by,
+            'limit': self._limit,
+            'offset': self._offset,
+        })
+
+    @property
+    def queryhelp(self) -> QueryDict:
+        """"Legacy name for ``as_dict`` method."""
+        # deprecate?
```

Comment:
The only outstanding question, for me, is whether we actually add a deprecation warning here, or "schedule" it for a later date. I guess essentially it shouldn't be removed until v3?

Reply:
I don't think it will be good for users to have deorecation warnings showing up when they've just switched to v2, so I would be tempted to wait